edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeHubDockerModule.cs

```diff
@@ -36,5 +36,28 @@ public class EdgeHubDockerModule : DockerModule, IEdgeHubModule
             DefaultValueHandling = DefaultValueHandling.Populate)]
         [DefaultValue(RestartPolicy.Always)]
         public override RestartPolicy RestartPolicy { get; }
+
+        public override bool Equals(IModule<DockerConfig> other)
```

Comment:
To share Equals logic across base (DockerModule) and derived classes (EdgeAgentDockerModule, EdgeHubDockerModule), you may need to extract some basic fields in another method e.g. BasicEquals or CoreEquals.

Reply:
Why doens't base Equals work here? It doesnt' compare versions either.

Reply:
@philipktlin 
The edgeAgent already has its own equal implementation that has been ignoring the versioning. 

@varunpuranik 
The base's Equal() check the `this.Version` as part of the equal. We don't need the version checking here for edgeHub.